### PR TITLE
Add support for Point and MultiPoint geometries in shape_bbox_indexer

### PR DESCRIPTION
**Pull request by huard**
### Pull Request Checklist:
- [x] This PR addresses an already opened issue (for bug fixes / features)
    - This PR fixes issue #283
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Documentation has been added / updated (for bug fixes / features)
- [x] HISTORY.rst has been updated (with summary of main changes)
- [x] I have added my relevant user information to `AUTHORS.md`

* **What kind of change does this PR introduce?:** 
Feature

* **Does this PR introduce a breaking change?:** 
No

* **Other information:**

### CODE REVIEW

File: tests/core/test_subset.py [hunk 1]
```diff
@@ -1074,6 +1074,24 @@ def test_rectilinear(self):
             inds = subset.shape_bbox_indexer(ds, gpd.GeoDataFrame(geometry=[pb]))
             assert pb.within(subset.grid_exterior_polygon(ds.isel(inds)))
 
+    def test_complex_geometries(self):
+        """Test with geometries that cannot be simplified to a single polygon using `unary_union`."""
+        import shapely.wkt
```
Comment (Zeitsperre) [inline tests/core/test_subset.py:1079]:
```suggestion
        pytest.importorskip("xesmf", "0.6.2")
        import shapely.wkt
```

**Comment by Zeitsperre** [comment]
Memo: I need to add a step to the CI for outputting the versions of pip libraries. This already exists in other workflows. Will add here.

File: clisops/core/subset.py [hunk 2]
```diff
@@ -771,34 +771,52 @@ def shape_bbox_indexer(ds, poly):
...
+    rectilinear = is_rectilinear(ds)
+    if rectilinear:
+        hull = hull.minimum_rotated_rectangle
+
+    if hasattr(hull.boundary, "geoms"):
+        # Handle cases with 1 point. Putting the same point twice to avoid issues with zip later.
+        if len(hull.boundary.geoms) == 0:
+            coords = hull.xy, hull.xy
+        # Handle cases with 2 points
+        elif len(hull.boundary.geoms) <= 2:
```
Comment (aulemahal) [inline clisops/core/subset.py:802]: What happens with `len(...) > 2` ?

**Review by aulemahal — APPROVED** [review]
This looks good. 

While testing I noticed that when using curvilinear grids that do not have bounds included, this fails with `'Dataset' object has no attribute 'lon_b'`. Three thoughts:
- This error message is a bit cryptic, something like "bounds are missing" would make more sense.
- `cf-xarray` can now estimate those bounds (xESMF uses this)
- Are the bounds totally necessary ? With the halo included, couldn't this revert back to using the centers only ?

File: clisops/core/subset.py [hunk 3]
```diff
@@ -771,34 +771,52 @@ def shape_bbox_indexer(ds, poly):
...
+        hull = geom.convex_hull.unary_union.convex_hull
+    elif isinstance(poly, gpd.array.GeometryArray):
+        hull = poly.convex_hull.unary_union().convex_hull
+    elif isinstance(poly, (list, tuple)):
+        hpoly = [p.convex_hull for p in poly]
+        hull = unary_union(hpoly).convex_hull
 
     # If polygon sits on the grid boundary, we need to roll the grid's coordinates and this is not supported.
-    if not grid_exterior_polygon(ds).contains(envelope):
+    if not grid_exterior_polygon(ds).contains(hull):
```
[the remark this answers is not in the document]
Comment (huard) [reply clisops/core/subset.py:789]: I was concerned it would create noise for "just an optimization".

File: clisops/core/subset.py (hunk 2 above)
Comment (huard) [reply clisops/core/subset.py:802]: Then it's a polygon and it should be handled by the else below. But I see your point, I could replace the elif by an else.

**Review by Zeitsperre — APPROVED** [review]

**Pull request merged**